packages/react-storage/src/components/StorageBrowser/views/LocationsView/useLocationsView.ts

```diff
@@ -71,18 +67,25 @@ export function useLocationsView(
   }, [handleList, listOptions]);
 
   // set up pagination
-  const onPaginateNext = () =>
+  const onPaginate = () => {
+    if (!nextToken) return;
     handleList({
       options: { ...listOptions, nextToken },
     });
+  };
 
   const {
     currentPage,
-    handlePaginateNext,
-    handlePaginatePrevious,
+    handlePaginate,
     handleReset,
+    highestPageVisited,
     range,
-  } = usePaginate({ onPaginateNext, pageSize: listOptions.pageSize });
+  } = usePaginate({
+    onPaginate,
+    pageSize: listOptions.pageSize,
+    resultCount,
+    hasNextToken,
+  });
```

Comment:
Would like to see `items` passed to `usePaginate` so it can return `pageItems` directly but i think @pranavosu may be handling that already